### PR TITLE
Per upstream connection settings

### PR DESCRIPTION
**Pull request by yuval-k**
Solves part of https://github.com/solo-io/gloo/issues/668 - allow configuration of connection settings.

### CODE REVIEW

File: projects/gloo/pkg/plugins/upstreamconn/plugin.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	return &Plugin{}
+}
+
+func (p *Plugin) Init(params plugins.InitParams) error {
+	return nil
+}
+
+func (p *Plugin) ProcessUpstream(params plugins.Params, in *v1.Upstream, out *envoyapi.Cluster) error {
+
+	cfg := in.GetUpstreamSpec().GetConnectionConfig()
```
Comment (EItanya) [inline projects/gloo/pkg/plugins/upstreamconn/plugin.go]: This can be nil, worried about panics?

Comment (yuval-k) [reply projects/gloo/pkg/plugins/upstreamconn/plugin.go]: not when using getters

**Pull request merged**